tiledb/sm/c_api/tiledb.cc

```diff
@@ -3974,6 +3975,43 @@ int32_t tiledb_serialize_array_max_buffer_sizes(
   return TILEDB_OK;
 }
 
+capi_return_t tiledb_deserialize_fragments_timestamps(
+    tiledb_serialization_type_t serialize_type,
+    const tiledb_buffer_t* buffer,
+    uint64_t* timestamp_start,
+    uint64_t* timestamp_end) noexcept {
+  api::ensure_buffer_is_valid(buffer);
+  api::ensure_output_pointer_is_valid(timestamp_start);
+  api::ensure_output_pointer_is_valid(timestamp_end);
+
+  // Deserialize
+  tie(*timestamp_start, *timestamp_end) =
+      tiledb::sm::serialization::fragments_timestamps_deserialize(
+          (tiledb::sm::SerializationType)serialize_type, buffer->buffer());
+
+  return TILEDB_OK;
+}
+
+capi_return_t tiledb_deserialize_fragments_list(
+    const char* array_uri,
+    tiledb_serialization_type_t serialize_type,
+    const tiledb_buffer_t* buffer,
+    tiledb_fragments_list_t** fragments) noexcept {
+  api::ensure_buffer_is_valid(buffer);
+  api::ensure_output_pointer_is_valid(fragments);
+
+  // Deserialize
+  std::vector<tiledb::sm::URI> uris =
+      tiledb::sm::serialization::fragments_list_deserialize(
+          tiledb::sm::URI(array_uri),
+          (tiledb::sm::SerializationType)serialize_type,
+          buffer->buffer());
+
+  *fragments = tiledb_fragments_list_handle_t::make_handle(uris);
```

Comment:
Can you please investigate for a few minutes if it's complex to move this fragment uris vector instead of copying it around? My intuition is that it's just a vector&& constructor in `fragments_list_handle_t`